addons/website_blog/static/src/js/website_blog.js

```diff
@@ -99,4 +104,38 @@ publicWidget.registry.websiteBlog = publicWidget.Widget.extend({
         dom.scrollTo($el[0], {duration: duration}).then(() => callback());
     },
 });
+
+/**
+ * Adjusts the containers'width class based on the first (text) section of the
+ * blog post content.
+ * If there is a text section it uses the first text section, otherwise it
+ * uses the first section.
+ *
+ * @static
+ * @param {Element} blogPostContentEl blog post content element
+ * @param {Element} containerEl container element to adjust
+ */
+publicWidget.registry.websiteBlog.adjustWidth = function (blogPostContentEl, containerEl) {
```

Comment:
Put the method back inside the widget, and made the call from wysiwyg through a `bus.trigger()`.
(Cannot rely solely on reacting to options change because reordering snippets is not such a change - but impacts which text block is the first one.)